src/v/redpanda/application.h

```diff
@@ -91,7 +91,7 @@ inline const auto redpanda_start_time{
   std::chrono::duration_cast<std::chrono::milliseconds>(
     std::chrono::system_clock::now().time_since_epoch())};
 
-class application {
+class application : public ssx::sharded_service_container {
```

Comment:
this changes the ordering, right? before it was deferred things via the loop in ~application then all the members, now it is all the members, and the deferred things via the loop in ~sharded_service_container. does that matter? i just raise the point because we've had a long history of fine tuning the application shutdown stuff.

if there is an issue, i guess the simplest option would be to embed the container instead of inheriting from it?

Reply:
yeh i'm good with leaving it. we'll get a good couple 10K+ shutdowns by the time we release next :P